src/wrappers/pubsub.js

```diff
@@ -33,8 +34,11 @@ function pubSubSubscriberMiddleware(message, originalHandler, requestFunctionThi
         let triggerMetadata = { messageId: message.id };
         pubSubEvent.setId(message.id);
         const messageData = (message.data && JSON.parse(`${message.data}`));
-        if (messageData && typeof messageData === 'object') {
-            triggerMetadata = Object.assign(triggerMetadata, messageData);
+        // add message only when METADATA_ONLY === FALSE
+        if (!config.getConfig().metadataOnly) {
+            if (messageData && typeof messageData === 'object') {
+                triggerMetadata = Object.assign(triggerMetadata, messageData);
+            }
```

Comment:
why not use the third parameter to `addMetadata`? i'd rather the instrumentation not get the config

Reply:
fixed